src/app/components/editors/quest/select-quest/select-quest.integration.spec.ts

```diff
@@ -0,0 +1,203 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { RouterTestingModule } from '@angular/router/testing';
+import { Router } from '@angular/router';
+import { of } from 'rxjs';
+import Spy = jasmine.Spy;
+
+import { QueryService } from '../../../../services/query.service';
+import { SelectQuestComponent } from './select-quest.component';
+import { QuestSelectService } from '../../../../services/select/quest-select.service';
+import { SelectQuestModule } from './select-quest.module';
+import { PageObject } from '../../../../test-utils/page-object';
+import { QuestTemplate } from '../../../../types/quest-template.type';
+
+class SelectQuestComponentPage extends PageObject<SelectQuestComponent> {
+  get createInput() { return this.query<HTMLInputElement>('app-create input#id'); }
+  get selectNewBtn() { return this.query<HTMLButtonElement>('app-create #select-button'); }
+  get freeStatusWrapper() { return this.query<HTMLDivElement>('#id-free-status'); }
+
+  get queryWrapper() { return this.query<HTMLElement>('code.hljs'); }
+
+  get searchIdInput() { return this.query<HTMLInputElement>('input#search-id'); }
+  get searchNameInput() { return this.query<HTMLInputElement>('input#name'); }
+  get searchLimitInput() { return this.query<HTMLInputElement>('input#limit'); }
+  get searchBtn() { return this.query<HTMLButtonElement>('#search-btn'); }
+
+  get topBar() { return this.query<HTMLElement>('app-top-bar'); }
+
+  expectTopBarCreatingNew(id: number) {
+    expect(this.topBar.innerText).toContain(`Creating new: ${id}`);
+  }
+
+  expectTopBarEditing(id: number, name: string) {
+    expect(this.topBar.innerText).toContain(`Editing: ${name} (${id})`);
+  }
+
+  expectNewEntityFree() {
```

Comment:
the functions around here could be moved in the super class
- expectTopBarCreatingNew
- expectTopBarEditing
- expectNewEntityFree
- expectEntityAlreadyInUse

Reply:
done https://github.com/azerothcore/Keira3/pull/34/commits/5446cc8a1b35e39b150a9df6f76c659c0db1dd49